demo-app/src/maplibreNativeMain/kotlin/dev/sargunv/maplibrecompose/demoapp/demos/OfflineDemo.kt

```diff
@@ -0,0 +1,108 @@
+import androidx.compose.material3.Button
+import androidx.compose.material3.ListItem
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalDensity
+import dev.sargunv.maplibrecompose.compose.MaplibreMap
+import dev.sargunv.maplibrecompose.compose.offline.DownloadState
+import dev.sargunv.maplibrecompose.compose.offline.OfflineRegionDefinition
+import dev.sargunv.maplibrecompose.compose.offline.OfflineRegionStatus
+import dev.sargunv.maplibrecompose.compose.offline.rememberOfflineRegionManager
+import dev.sargunv.maplibrecompose.compose.rememberCameraState
+import dev.sargunv.maplibrecompose.compose.rememberStyleState
+import dev.sargunv.maplibrecompose.core.CameraPosition
+import dev.sargunv.maplibrecompose.demoapp.DEFAULT_STYLE
+import dev.sargunv.maplibrecompose.demoapp.Demo
+import dev.sargunv.maplibrecompose.demoapp.DemoMapControls
+import dev.sargunv.maplibrecompose.demoapp.DemoOrnamentSettings
+import dev.sargunv.maplibrecompose.demoapp.DemoScaffold
+import kotlinx.coroutines.launch
+
+object OfflineDemo : Demo {
+  override val name: String
+    get() = "Offline regions"
+
+  override val description: String
+    get() = "Save regions of the map to device storage."
+
+  @Composable
+  override fun Component(navigateUp: () -> Unit) {
+    DemoScaffold(this, navigateUp) {
+      Column {
+        val cameraState = rememberCameraState(firstPosition = CameraPosition())
+        val styleState = rememberStyleState()
+
+        Box(modifier = Modifier.weight(0.5f)) {
+          MaplibreMap(
+            styleUri = DEFAULT_STYLE,
+            cameraState = cameraState,
+            styleState = styleState,
+            ornamentSettings = DemoOrnamentSettings(),
+          )
+          DemoMapControls(cameraState, styleState)
+        }
+
+        Column(modifier = Modifier.weight(0.5f)) {
+          val coroutineScope = rememberCoroutineScope()
+          val offlineManager = rememberOfflineRegionManager()
+          val density = LocalDensity.current
+
+          Button(
+            onClick = {
+              coroutineScope.launch {
+                offlineManager
+                  .create(
+                    OfflineRegionDefinition.TilePyramid(
+                      styleUrl = DEFAULT_STYLE,
+                      bounds = cameraState.awaitProjection().queryVisibleBoundingBox(),
+                      pixelRatio = density.density,
+                    )
+                  )
+                  .setDownloadState(DownloadState.Active)
+              }
+            },
+            enabled = cameraState.position.zoom >= 10.0,
+          ) {
+            Text("Save region")
+          }
+
+          Column(modifier = Modifier.verticalScroll(rememberScrollState())) {
```

Comment:
The demo is currently quite ugly, but I'd rather polish it up in a future PR (thinking some sort of bottom sheet containing controls and list, draggable up over the map or down out of the way)

Reply:
Nvm, went ahead and redid the demo in this PR